.github/workflows/tardis-research-papers.yml

```diff
@@ -0,0 +1,117 @@
+name: tardis-research-papers
+
+on:
+  schedule:
+    - cron: "0 0 1 * *"
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+env:
+  NASA_ADS_TOKEN: ${{ secrets.NASA_ADS_TOKEN }}
+
+jobs:
+    research-using-tardis:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+
+            - name: Setup Python
+              uses: actions/setup-python@v5
+              with:
+                python-version: "3.x"
+    
+            - name: Install dependencies
+              run: |
+                python -m pip install --upgrade pip
+                pip install jupyter nbconvert
+                        
+            - name: Run the ADS notebook
+              run: |
+                jupyter nbconvert --to python  docs/resources/research_done_using_TARDIS/ads.ipynb
+                python3 docs/resources/research_done_using_TARDIS/ads.py
+            
+            - uses: actions/upload-artifact@v4
+              with:
+                name: research_papers
+                path: |
+                    resource_papers.rst
+            
+    pull_request:
+        needs: [research-using-tardis]
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+
+            - uses: actions/download-artifact@v4
+              with:
+                path: /tmp
+    
+            - name: Copy files to repository
+              run: |
+                cp -r /tmp/resource_papers/* docs/resources/research_done_using_TARDIS/
+    
+            - name: Get current date
+              run: echo "DATE=$(date +'%Y.%m.%d')" >> $GITHUB_ENV
+    
+            - uses: peter-evans/create-pull-request@v4
```

Comment:
these actions usually need a pull_request_target trigger- have you tested job this on your fork?

Reply:
you can check the comments, I have posted my sample runs there